pkg/apis/build/v1alpha2/build.go

```diff
@@ -137,8 +137,7 @@ func (b *Build) PodName() string {
 }
 
 func (b *Build) MetadataReady(pod *corev1.Pod) bool {
-	return !b.Status.GetCondition(corev1alpha1.ConditionSucceeded).IsTrue() &&
-		pod.Status.Phase == "Succeeded"
+	return pod.Status.Phase == corev1.PodSucceeded || (pod.Status.Phase == corev1.PodFailed && pod.Status.Reason == "DeadlineExceeded")
```

Comment:
is it possible for deadlineexceeded to be an actual error?

Reply:
Yes, but since we validate that the completion step had finished, the build should be considered successful. If the build exits with deadline exceeded and the completion step hadnt finished then it will still fail